File: ref/algorithmic-trading/packages/shared/src/types/position.ts
/** 보유 포지션 */
export interface Position {
  id: number;
  stockCode: string;
  stockName: string;
  quantity: number;
  avgPrice: number;
  currentPrice: number;
  pnl: number;
  pnlRate: number;
  boughtAt: string;
  updatedAt: string;
}

/** 포트폴리오 스냅샷 (일별) */
export interface PortfolioSnapshot {
  id: number;
  date: string;
  totalValue: number;
  cash: number;
  stockValue: number;
  dailyPnl: number;
  dailyPnlRate: number;
  cumulativePnlRate: number;
  createdAt: string;
}

File: ref/algorithmic-trading/apps/app/app/src/main/java/com/trading/app/domain/model/Position.kt
package com.trading.app.domain.model

data class Position(
    val id: Long = 0,
    val stockCode: String,
    val stockName: String,
    val quantity: Int,
    val avgPrice: Double,
    val currentPrice: Double,
    val pnl: Double,
    val pnlRate: Double,
    val boughtAt: String,
    val updatedAt: String,
    val strategyId: String = "A",
    val source: String = "engine",
)

data class PortfolioSnapshot(
    val id: Long = 0,
    val date: String,
    val totalValue: Double,
    val cash: Double,
    val stockValue: Double,
    val dailyPnl: Double,
    val dailyPnlRate: Double,
    val cumulativePnlRate: Double,
    val createdAt: String,
)

data class PortfolioContext(
    val cash: Double,
    val totalEquity: Double,
    val currentPositionCount: Int,
)
